utils/misc.js

```diff
@@ -121,6 +123,34 @@ export default class Miscellaneous {
         return crypto.createHash('md5').update(data).digest('hex');
     }
 
+    /**
+     * Validate if the incoming webhook contains valid secret key.
+     *
+     * @param {express.Request} request - The express request object.
+     * @return {Promise<boolean>} True if valid, false otherwise.
+     */
+    static async isPostSecure(request) {
+        const payload = JSON.stringify(request.body);
+        const ghostConfigs = await ProjectConfigs.ghost();
+        const signatureWithDate = request.headers['x-ghost-signature'];
+
+        // Secret set on Ghosler but not recd. in the request headers.
+        if (ghostConfigs.secret && !signatureWithDate) {
+            logError(logTags.Express, 'The \'X-Ghost-Signature\' header not found in the request. Did you setup the Secret Key correctly?');
+            return false;
+        }
+
+        // @see: https://github.com/TryGhost/Ghost/blob/efb2b07c601cd557976bcbe12633f072da5c22a7/ghost/core/core/server/services/webhooks/WebhookTrigger.js#L98
+        const signature = signatureWithDate.split(', ')[0].replace('sha256=', '');
```

Comment:
Should we also compare the timestamp received in the signature?
Doesn't hurt to add another check though. `5 minutes` should be more than enough here.

Reply:
Done.